### PR TITLE
860 adding autoLogin documentation

### LINKED ISSUE #860: fix Form Fields section from Register documentation
since stormpath.web.register.form.fields is not valid anymore adjust this section from both Servlet documentation and Spring Boot documentation

### PR DESCRIPTION
**Pull request by mzumbadoate**
resolves #860

### CODE REVIEW

File: docs/source/registration.rst [hunk 1]
```diff
@@ -59,13 +68,27 @@ Again, this functionality is only executed if email verification is disabled.  I
...
+    stormpath.web.register.form.fields.givenName.placeholder = stormpath.web.register.form.fields.givenName.placeholder
+    stormpath.web.register.form.fields.givenName.required = true
+    stormpath.web.register.form.fields.givenName.type = text
+
+By default the supported directives are:
 
-The value is a comma-delimited list Stormpath ``Account`` field names and optional form field directives.  The currently supported field names:
+* ``enabled``: the field is enabled for edit.
+* ``visible``: the field is visible.
+* ``label``: this is the label to display and it points to the i18n properties as ``stormpath.web.register.form.fields.givenName.label``.
```
Comment (mraible) [inline docs/source/registration.rst]: I would reword this as "this is the label to display and its value is defined in ``i18n.properties`` with the key ``stormpath.web.register.form.fields.givenName.label``.

File: docs/source/registration.rst [hunk 2]
```diff
@@ -59,13 +68,27 @@ Again, this functionality is only executed if email verification is disabled.  I
...
+* ``placeholder``: this is the placeholder to display and it points to the i18n properties as ``stormpath.web.register.form.fields.givenName.placeholder``.
```
Comment (mraible) [inline docs/source/registration.rst]: this is the placeholder to display and its value is defined in ``i18n.properties`` as ``stormpath.web.register.form.fields.givenName.placeholder``.

File: docs/source/registration.rst [hunk 3]
```diff
@@ -59,13 +68,27 @@ Again, this functionality is only executed if email verification is disabled.  I
...
+* ``required``: makes the field require, if it is required and not provided the error message is in the i18n properties as ``stormpath.web.register.form.fields.email.required``. If the input is invalid the error message is in the i18n properties as ``stormpath.web.register.form.fields.email.invalid``.
```
Comment (mraible) [inline docs/source/registration.rst]: Change to "makes the field required. If the field is blank, the error message is defined in ``i18n.properties`` as ``stormpath.web.register.form.fields.email.required``. If the input is invalid, the error message is defined as ``stormpath.web.register.form.fields.email.invalid``.

File: extensions/spring/boot/docs/source/registration.rst [hunk 4]
```diff
@@ -36,12 +36,20 @@ If you want to change this path, set the ``stormpath.web.register.uri`` configur
...
+**If** ``stormpath.web.register.autoLogin = false`` **, this is the default**
 
 If :ref:`email verification <email verification>` is disabled, a successfully registered user will be automatically redirected to the application's context root (home page) by default.  If you want to change this destination, set the ``stormpath.web.register.nextUri`` configuration property:
 
 .. code-block:: properties
 
     stormpath.web.register.nextUri = /
+**If** ``stormpath.web.register.autoLogin = true``
+
+If :ref:`email verification <email verification>` is disabled, a successfully registered user will be automatically logged in to the application and redirected to the page a successful login redirects, configuration property:
```
Comment (mraible) [inline extensions/spring/boot/docs/source/registration.rst]: I'd suggest changing this to "redirected to the page specified in the ``stormpath.web.login.nextUri`` configuration property:"

File: extensions/spring/boot/docs/source/registration.rst [hunk 5]
```diff
@@ -70,13 +78,27 @@ Remember that the property value is the *name* of a view, and the effective Spri
...
+    stormpath.web.register.form.fields.givenName.placeholder = stormpath.web.register.form.fields.givenName.placeholder
+    stormpath.web.register.form.fields.givenName.required = true
+    stormpath.web.register.form.fields.givenName.type = text
 
-The value is a comma-delimited list Stormpath ``Account`` field names and optional form field directives.  The currently supported field names:
+By default the supported directives are:
+
+* ``enabled``: the field is enabled for edit.
+* ``visible``: the field is visible.
+* ``label``: this is the label to display and it points to the i18n properties as ``stormpath.web.register.form.fields.givenName.label``.
```
Comment (mraible) [inline extensions/spring/boot/docs/source/registration.rst]: Same comments as above for `docs/source/registration.rst`

File: extensions/spring/boot/docs/source/registration.rst [hunk 6]
```diff
@@ -85,30 +107,9 @@ The value is a comma-delimited list Stormpath ``Account`` field names and option
...
-
-1. The ``givenName`` form field (first name) will be shown first, but it is optional (no directives)
-2. The ``surname`` form field (last name) will be shown next, but it is optional (no directives)
-3. The ``email`` form field will be shown next and it is required (``required`` directive)
-4. The ``password`` form field will be shown last, and it is required and a password field (show stars instead of raw characters).
-
-You can customize this list with optional directives as necessary, but note:
-
-**The** ``email`` **form field is always required.  If you customize your form fields, ensure that you always have at least an** ``email(required)`` **list entry.**
+You can customize this list of fields, those extra fields will be saved as custom data. Also you can customize additional directives as necessary, but note:
```
Comment (mraible) [inline extensions/spring/boot/docs/source/registration.rst]: What about customizing the order of the fields with the following property?

`stormpath.web.register.form.fieldOrder = username,givenName,middleName,surname,email,password,confirmPassword`

**Comment by mraible** [comment]
@mzumbadoate I've added comments about some of the wording. Back to you.

**Comment by mzumbadoate** [comment]
@mraible PR updated

File: docs/source/registration.rst [hunk 7]
```diff
@@ -59,13 +68,27 @@ Again, this functionality is only executed if email verification is disabled.  I
...
+    stormpath.web.register.form.fields.givenName.required = true
+    stormpath.web.register.form.fields.givenName.type = text
+
+By default the supported directives are:
+
+* ``enabled``: the field is enabled for edit.
+* ``visible``: the field is visible.
+* ``label``: this is the label to display and its value is defined in i18n.properties with the key ``stormpath.web.register.form.fields.givenName.label``.
+* ``placeholder``: this is the placeholder to display and its value is defined in i18n.properties with the key ``stormpath.web.register.form.fields.givenName.placeholder``.
+* ``required``: makes the field required, If the field is blank, the error message is defined in i18n.properties as``stormpath.web.register.form.fields.email.required``. If the input is invalid the error message is in the i18n.properties with the key ``stormpath.web.register.form.fields.email.invalid``.
```
Comment (mraible) [inline docs/source/registration.rst]: Missing a space after "i18n.properties as"

File: docs/source/registration.rst [hunk 8]
```diff
@@ -59,13 +68,27 @@ Again, this functionality is only executed if email verification is disabled.  I
...
+    stormpath.web.register.form.fields.givenName.label = stormpath.web.register.form.fields.givenName.label
+    stormpath.web.register.form.fields.givenName.placeholder = stormpath.web.register.form.fields.givenName.placeholder
+    stormpath.web.register.form.fields.givenName.required = true
+    stormpath.web.register.form.fields.givenName.type = text
+
+By default the supported directives are:
+
+* ``enabled``: the field is enabled for edit.
+* ``visible``: the field is visible.
+* ``label``: this is the label to display and its value is defined in i18n.properties with the key ``stormpath.web.register.form.fields.givenName.label``.
```
Comment (mraible) [inline docs/source/registration.rst]: Do we highlight field names with back ticks? For example, `i18n.properties`. Not sure, asking for the convention.

File: extensions/spring/boot/docs/source/registration.rst [hunk 9]
```diff
@@ -70,13 +79,27 @@ Remember that the property value is the *name* of a view, and the effective Spri
...
+    stormpath.web.register.form.fields.givenName.required = true
+    stormpath.web.register.form.fields.givenName.type = text
+
+By default the supported directives are:
+
+* ``enabled``: the field is enabled for edit.
+* ``visible``: the field is visible.
+* ``label``: this is the label to display and its value is defined in i18n.properties with the key ``stormpath.web.register.form.fields.givenName.label``.
+* ``placeholder``: this is the placeholder to display and its value is defined in i18n.properties with the key ``stormpath.web.register.form.fields.givenName.placeholder``.
+* ``required``: makes the field required, If the field is blank, the error message is defined in i18n.properties as``stormpath.web.register.form.fields.email.required``. If the input is invalid the error message is in the i18n.properties with the key ``stormpath.web.register.form.fields.email.invalid``.
```
Comment (mraible) [inline extensions/spring/boot/docs/source/registration.rst]: Missing space before ``stormpath.web.register.form.fields.email.required``

**Comment by mraible** [comment]
@mzumbadoate Added a couple comments about missing spaces and wondering how filenames are handled with back ticks.

**Comment by mraible** [comment]
LGTM 👍 pending CI success
